mlrun/serving/routers.py

```diff
@@ -157,3 +163,256 @@ def _handle_event(self, event):
+        Supports:
+        - <prefix>/<model>[/versions/<ver>]/operation
+        Sends the event to the specific <model>[/versions/<ver>]
+
+        - <prefix>/operation
+        Sends the event to all models
+
+        Args:
+            body (dict): event body
+            urlpath (string): url path
+
+        Raises:
+            ValueError: [description]
+
+        Returns:
+            [type]: [description]
+        """
+        subpath = None
+        model = ""
+        if urlpath and not urlpath == "/":
+            # process the url <prefix>/<model>[/versions/<ver>]/operation
+            subpath = ""
+            urlpath = urlpath[len(self.url_prefix) :].strip("/")
+            if not urlpath:
+                # Only self.url_prefix was given
+                return "", None, ""
+            segments = urlpath.split("/")
+            if len(segments) == 1:
+                # Are we looking at a router level operation?
+                if segments[0] in ["infer", "predict", "explain"]:
+                    # We are given an operation
+                    return self.name, None, segments[0]
+            model = segments[0]
+            if len(segments) > 2 and segments[1] == "versions":
+                model = model + ":" + segments[2]
+                segments = segments[2:]
+            if len(segments) > 1:
+                subpath = "/".join(segments[1:])
+
+        if isinstance(body, dict):
+            # accepting route information from body as well
+            # to support streaming protocols (e.g. Kafka).
+            model = model or self.name
+            subpath = body.get("operation", subpath)
+        if subpath is None:
+            subpath = "infer"
+
+        if model in self.routes:
+            self.log_router = False
+        elif model != self.name:
+            models = " | ".join(self.routes.keys())
+            raise ValueError(
+                f"model {model} doesnt exist, available models: {models} or an operation alone for ensemble operation"
+            )
+        else:
+            self.log_router = True
+        return model, self.routes[model], subpath
+
+    def _max_vote(self, all_predictions):
+        """Returns an argmax"""
+        return [
+            max(predictions, key=predictions.count) for predictions in all_predictions
+        ]
+
+    def _mean_vote(self, all_predictions):
+        return [mean(predictions) for predictions in all_predictions]
+
+    def _is_int(self, n):
+        return float(n).is_integer()
+
+    def _vote(self, events):
+        if "outputs" in events.body:
+            # Dealing with a specific model prediction
+            return events
+        predictions = [
+            model.body["outputs"]["prediction"] for _, model in events.body.items()
+        ]
+
+        flattened_predictions = [
+            [predictions[j][i] for j in range(len(predictions))]
+            for i in range(len(predictions[0]))
+        ]
+        if self.vote_flag is None:
```

Comment:
`self.vote_flag` is never `None`, it's either `True` or `False`

Reply:
changed to `not self.vote_flag`